src/stream-tcp.c

```diff
@@ -1065,22 +1065,26 @@ void StreamTcpSetOSPolicy(TcpStream *stream, Packet *p)
  *  \param stream stream to update
  *  \param ack ACK value to test and set
  */
-#define StreamTcpUpdateLastAck(ssn, stream, ack) { \
-    if (SEQ_GT((ack), (stream)->last_ack)) \
-    { \
-        SCLogDebug("ssn %p: last_ack set to %"PRIu32", moved %u forward", (ssn), (ack), (ack) - (stream)->last_ack); \
-        if ((SEQ_LEQ((stream)->last_ack, (stream)->next_seq) && SEQ_GT((ack),(stream)->next_seq))) { \
-            SCLogDebug("last_ack just passed next_seq: %u (was %u) > %u", (ack), (stream)->last_ack, (stream)->next_seq); \
-        } else { \
-            SCLogDebug("next_seq (%u) <> last_ack now %d", (stream)->next_seq, (int)(stream)->next_seq - (ack)); \
-        }\
-        (stream)->last_ack = (ack); \
-        StreamTcpSackPruneList((stream)); \
-    } else { \
-        SCLogDebug("ssn %p: no update: ack %u, last_ack %"PRIu32", next_seq %u (state %u)", \
-                    (ssn), (ack), (stream)->last_ack, (stream)->next_seq, (ssn)->state); \
-    }\
-}
+#define StreamTcpUpdateLastAck(ssn, stream, ack)                                                   \
+    {                                                                                              \
+        if (SEQ_GT((ack), (stream)->last_ack) && SEQ_GT(ack, (stream)->base_seq)) {                \
```

Comment:
Maybe we should just check `base_seq` (since`last_ack` is greater)

Original diff before clang-format
```diff
diff --git a/src/stream-tcp.c b/src/stream-tcp.c
index 9a72e22c16..6b419599be 100644
--- a/src/stream-tcp.c
+++ b/src/stream-tcp.c
@@ -1066,7 +1066,7 @@ void StreamTcpSetOSPolicy(TcpStream *stream, Packet *p)
  *  \param ack ACK value to test and set
  */
 #define StreamTcpUpdateLastAck(ssn, stream, ack) { \
-    if (SEQ_GT((ack), (stream)->last_ack)) \
+    if (SEQ_GT((ack), (stream)->last_ack) && SEQ_GT(ack, (stream)->base_seq)) \
     { \
         SCLogDebug("ssn %p: last_ack set to %"PRIu32", moved %u forward", (ssn), (ack), (ack) - (stream)->last_ack); \
```

Reply:
> Maybe we should just check base_seq (sincelast_ack is greater)

Makes sense to me.

> Original diff before clang-format

Thanks for this. 😅